src/Sylius/Component/Channel/Context/CachedPerRequestChannelContext.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Component\Channel\Context;
+
+use Sylius\Component\Channel\Model\ChannelInterface;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+/**
+ * @author Kamil Kokot <kamil.kokot@lakion.com>
+ */
+final class CachedPerRequestChannelContext implements ChannelContextInterface
+{
+    /**
+     * @var ChannelContextInterface
+     */
+    private $decoratedChannelContext;
+
+    /**
+     * @var RequestStack
+     */
+    private $requestStack;
+
+    /**
+     * @var \SplObjectStorage|ChannelInterface[]
+     */
+    private $requestToChannelMap;
+
+    /**
+     * @var \stdClass
+     */
+    private $nullObject;
+
+    /**
+     * @param ChannelContextInterface $decoratedChannelContext
+     * @param RequestStack $requestStack
+     */
+    public function __construct(ChannelContextInterface $decoratedChannelContext, RequestStack $requestStack)
+    {
+        $this->decoratedChannelContext = $decoratedChannelContext;
+        $this->requestStack = $requestStack;
+
+        $this->requestToChannelMap = new \SplObjectStorage();
+        $this->nullObject = new \stdClass();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getChannel()
+    {
+        $objectIdentifier = $this->requestStack->getMasterRequest() ?: $this->nullObject;
```

Comment:
Why do we need reference to this null object? Won't it be better to just create new `\stdClass()` instance here?